Mathlib/Algebra/DirectSum/Internal.lean

```diff
@@ -276,6 +276,56 @@ theorem coe_of_mul_apply {i : ι} (r : A i) (r' : ⨁ i, A i) (n : ι) [Decidabl
 
 end CanonicallyOrderedAddCommMonoid
 
+section GradeZeroSubsemiring
+
+variable [Semiring R]
+variable [CanonicallyOrderedAddCommMonoid ι]
+variable [SetLike σ R] [AddSubmonoidClass σ R]
+variable (A : ι → σ) [SetLike.GradedMonoid (fun i ↦ A i)]
```

Comment:
```suggestion
variable (A : ι → σ) [SetLike.GradedMonoid A]
```
(untested)

Reply:
I've changed it to `SetLike.GradedMonoid A`.